calyx-frontend/src/ast.rs

```diff
@@ -14,8 +14,10 @@ pub struct NamespaceDef {
     pub components: Vec<ComponentDef>,
     /// Extern statements and any primitive declarations in them.
     pub externs: Vec<(Option<PosString>, Vec<Primitive>)>,
-    /// Optional opaque metadata
+    /// Optional legacy opaque metadata. Newer metadata is in [NamespaceDef::file_info_table]
     pub metadata: Option<String>,
+    /// Optional metadata mapping table
+    pub file_info_table: Option<SourceInfoTable>,
```

Comment:
Please change the comment to reflect our new naming: `/// Optional source info table`